### PR TITLE
fix: remove any duplicate statements in policy input

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: remove any duplicate statements in policy input

## Motivation and Context
Add support for removing duplicate statements automatically

## How to test this PR?
Unit tests added already you can try with `mc admin policy set` as well 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [x] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#e370cb92-dc17-49b5-b5dd-ca4e5d5a82e4) |
| mint-gateway-nas.sh | :x: [more...](#c7bf95be-cfb3-4a14-aa2a-fbd8f5396b55) |
<a name="c7bf95be-cfb3-4a14-aa2a-fbd8f5396b55"></a>
### 9385-5bbff80/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:30680
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1d8d72a914ef:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 8 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 31 seconds
(8/15) Running minio-dotnet tests ... done in 29 seconds
(9/15) Running minio-go tests ... FAILED in 32 seconds
{}

Executed 8 out of 15 tests successfully.
```
<a name="e370cb92-dc17-49b5-b5dd-ca4e5d5a82e4"></a>
### 9385-5bbff80/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:32072
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp dfcd5099c067:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 4 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... FAILED in 51 seconds
{
  "name": "awscli",
  "duration": 4650,
  "function": "aws --endpoint-url http://minio-dev3.minio.io:32072 s3api head-object --bucket awscli-mint-test-bucket-6253 --key datafile-1-kB-copy\n",
  "status": "FAIL",
  "error": "StorageClass was not applied"
}

Executed 4 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/utils.go [hunk 1]
```diff
@@ -471,7 +471,6 @@ func newCustomHTTPTransport(tlsConfig *tls.Config, dialTimeout time.Duration) fu
 		DialContext:           newCustomDialContext(dialTimeout, 15*time.Second),
 		MaxIdleConnsPerHost:   16,
 		MaxIdleConns:          16,
-		MaxConnsPerHost:       64, // This is used per drive/rpc host. More requests will block until free.
```
Comment (cesnietor) [inline cmd/utils.go]: is this part of this fix?

Comment (harshavardhana) [reply cmd/utils.go]: no it will be removed @cesnietor sent as a different fix

Comment (cesnietor) [reply cmd/utils.go]: cool apart from that all LGTM

**Review by cesnietor — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#2a530c0a-2b02-4a40-9bb7-d572ee8301e4) |
| mint-gateway-nas.sh | :x: [more...](#c72f71e7-7498-4512-893b-1af4c2ca5aed) |
<a name="c72f71e7-7498-4512-893b-1af4c2ca5aed"></a>
### 9385-b2d0201/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:30338
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp da72e6b42982:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 5 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 35 seconds
(8/15) Running minio-dotnet tests ... done in 28 seconds
(9/15) Running minio-go tests ... FAILED in 32 seconds
{}

Executed 8 out of 15 tests successfully.
```
<a name="2a530c0a-2b02-4a40-9bb7-d572ee8301e4"></a>
### 9385-b2d0201/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c4.minio.io:30345
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4dbc3a0f7320:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 5 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... FAILED in 54 seconds
{
  "name": "awscli",
  "duration": 5039,
  "function": "aws --endpoint-url http://minio-c4.minio.io:30345 s3api head-object --bucket awscli-mint-test-bucket-30190 --key datafile-1-kB-copy\n",
  "status": "FAIL",
  "error": "StorageClass was not applied"
}

Executed 4 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**